src/dev/i18n/README.md

```diff
@@ -0,0 +1,170 @@
+# I18n Tools
+
+## Default messages extraction tool
+
+### Description
+
+The tool is used to extract default messages from all `*.{js, ts, jsx, tsx, html, handlebars, hbs, pug}` files in provided plugins directories to a JSON file.
+
+It uses Babel to parse code and build an AST for each file or a single JS expression if whole file parsing is impossible. The tool is able to validate, extract and match IDs, default messages and descriptions only if they are defined statically and together, otherwise it will fail with detailed explanation. That means one can't define ID in one place and default message in another, or use function call to dynamically create default message etc.
+
+### Examples and restrictions
+
+**Global restrictions**:
+
+The `defaultMessage` and `description` must be string literals, template literals w/o expressions or string-only concatenation expressions, anything else is not allowed. The `id` can be only a string literal.
+
+The `defaultMessage` must contain ICU references to all keys in the `values` and vice versa.
+
+The `description` is optional, `values` is optional too unless `defaultMessage` references to it.
+
+* **Angular (.html)**
+
+  * **Filter**
+
+    ```
+    {{ ::'plugin_namespace.message_id' | i18n: {
```

Comment:
**nit:**
I'd use camelcase in the examples as we use in real code to not confuse readers
```suggestion
  {{ ::'pluginNamespace.messageId' | i18n: {
```